joomla/joomla-cms
# [Deprecate _QQ_ 2] Save language overrides with escaped double quotes

### Summary of Changes

As discussed in https://github.com/joomla/joomla-cms/issues/13421
This PR changes language overrides save to ini method so we use escaped double quotes (`\"`) instead of `"_QQ_"`.
Also adds a new method to save the language strings to a ini file so it can be easily used when needed.

### Testing Instructions

- Apply patch
- Create a new language override from Extensions -> Languages -> Overrides for the en-GB admin language with some double quotes on it (`"`). Save it, load it again and confirm if ok.
- Confirm it's saved as escaped double quote in `/administrator/language/overrides/en-GB.override.ini`.  Example: `A test with <a href=\"https://joomla.org\" target=\"_blank\">Joomla</a>`
- Manually add an override in `/administrator/language/overrides/en-GB.override.ini` with `_QQ_` and escaped quotes. Example:
```ini
JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED="A second test with <a href="_QQ_"https://joomla.org"_QQ_" target=\"_blank\">Joomla</a>"
```
- Confirm language override works
Ex: add to template index.php
```php
// Server side message system
$app->enqueueMessage(JText::_('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED'), 'warning');

// Client side message system (click on the page to show the javascript message)
JText::script('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED');
$this->addScriptDeclaration("document.addEventListener('click', function () { Joomla.renderMessages({'error': [Joomla.JText._('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED')]}) });");

```

### Documentation Changes Required

None.
